src/vs/base/parts/storage/node/storage.ts

```diff
@@ -297,11 +299,16 @@ export class SQLiteStorageDatabase implements IStorageDatabase {
 		return new Promise((resolve, reject) => {
 			import('vscode-sqlite3').then(sqlite3 => {
 				const connection: IDatabaseConnection = {
-					db: new (this.logger.isTracing ? sqlite3.verbose().Database : sqlite3.Database)(path, error => {
+					db: new (this.logger.isTracing ? sqlite3.verbose().Database : sqlite3.Database)(path, this.options.intent ?? SQLiteStorageDatabase.DEFAULT_MODE, error => {
```

Comment:
As mentioned above, pass `undefined` instead of default mode to leave the default up to the underlying library.